talk/owt/sdk/base/win/mediacapabilities.h

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) <2020> Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OWT_BASE_WIN_MEDIACAPABILITIES_H_
+#define OWT_BASE_WIN_MEDIACAPABILITIES_H_
+
+#include <mutex>
+#include <vector>
+#ifdef OWT_USE_MSDK
+#include "base_allocator.h"
+#include "mfxvideo++.h"
+#include "talk/owt/sdk/base/win/msdkvideobase.h"
+#endif
+#include "owt/base/commontypes.h"
+#include "talk/owt/sdk/base/mediautils.h"
+
+// Utilty class for checking media codec hardware acceleration status
+// on current platform. if MediaSDK is enabled, will use MediaSDK for
+// the check. Otherwise(TODO) will use MediaFoundation.
+
+namespace owt {
+namespace base {
+
+struct VideoEncoderCapability {
+  owt::base::VideoCodec codec_type;
+  bool has_trusted_rate_controller = true;   ///< When set to false, frame dropper is on.
+  bool hardware_accelerated = false;         ///< Video encoding is hardware accelerated.
+  bool discreted_graphics = false;           ///< Encoder runs on discreted or integrated graphics.
+  bool low_power = true;                     ///< Encoder runs on VDEnc.
+  uint8_t max_temporal_layers = 1;           ///< Maximum supported temporal layers.
+  uint8_t max_spatial_layers = 1;            ///< Maximum supported spatial layers.
+  CodecSpecificInfoUnion codec_specific;
+  std::vector<BRCMode> supported_brc_modes;  ///< BRC modes supported.
+  std::vector<SamplingMode> sampling_modes;
+};
+
+enum class VideoResolutionMax : int {
```

Comment:
Please avoid using names for resolutions. Using width and height instead.

Reply:
done